### PR TITLE
Added image zooming.

### LINKED ISSUE #10: Add image zoom option
At now, there is no option to zoom image.

### PR DESCRIPTION
**Pull request by fantoro**
Let me know if anything needs fixing or if everything's fine and if I should squash the commits.

Fixes #10

### CODE REVIEW

**Comment by scottmc** [comment]
You might also add support for zooming in/out with the mouse wheel when over the image

**Comment by fantoro** [comment]
> You might also add support for zooming in/out with the mouse wheel when over the image

Should I add that as a mouse mapping option (like the file explorer/image view scrolling mapping options)?

**Comment by scottmc** [comment]
Experiment and see how different options work, then pick the one(s) you like best and then ask here for opinions.

**Comment by fantoro** [comment]
> Experiment and see how different options work, then pick the one(s) you like best and then ask here for opinions.

Alright I decided to go with the way how the rest of the scroll wheel mapping options are written.

**Comment by scottmc** [comment]
If no one else gets to this I will test it later tonight.  I'm in California, so that's 6+ hours from now.

**Review by owenca — CHANGES_REQUESTED** [review]

File: source/pictureViewer.h
Comment (fantoro) [reply source/pictureViewer.h]: The rest of the methods in the header only have the type written. Should I leave it as it is so it fits in with the rest of the code or should I do it like in your suggestion?

File: source/pictureViewer.cpp
Comment (fantoro) [reply source/pictureViewer.cpp]: I decided to slightly change that expression to `z >= 0 ? z : imgZoom` because setting the zoom to 0 sets the image to just 1x1 px and I don't think that would be very useful for viewing images. Let me know if I should use the one you gave me or if I should keep it that way.

File: source/pictureViewer.h
Comment (owenca) [reply source/pictureViewer.h]: Ok.

Comment (fantoro) [reply source/pictureViewer.h]: So should I just leave it?

File: source/pictureViewer.cpp
Comment (owenca) [inline source/pictureViewer.cpp]:
```suggestion
    else if (
```

File: source/pictureViewer.cpp
Comment (owenca) [inline source/pictureViewer.cpp]:
```suggestion
```

File: source/pictureViewer.cpp
Comment (owenca) [inline source/pictureViewer.cpp]:
```suggestion
```

File: source/pictureViewer.cpp
Comment (owenca) [inline source/pictureViewer.cpp]: Delete this line and add the following at the beginning of the function:
```
  if (thePic == NULL) return;
```

Comment (owenca) [reply source/pictureViewer.cpp]: Also change the two `if (mode == ...)` statements below to `else if`.

File: source/pictureViewer.h
Comment (owenca) [reply source/pictureViewer.h]: It's up to you.

**Review by owenca — CHANGES_REQUESTED** [review]
When zooming, the center of the image or the cropped portion doesn't stay at the center of the image pane.

File: source/pictureViewer.cpp
Comment (owenca) [reply source/pictureViewer.cpp]: Maybe something like this:
```
		if (z < 0.1 ||  z > 10)
			return;

		imgZoom = z;
```

**Comment by owenca** [comment]
Zooming in decelerates while zoom out accelerates. You probably want to multiply/divide rather than adding/subtracting when updating the zoom factor.

File: source/pictureViewer.cpp
Comment (owenca) [inline source/pictureViewer.cpp]:
```suggestion
	if (z < 0.1 || z > 10)
		return;

	imgZoom = z;
```

File: source/pictureViewer.h
Comment (owenca) [reply source/pictureViewer.h]: It's up to you.

**Comment by scottmc** [comment]
Please fix this one up so we can merge soon.

**Comment by fantoro** [comment]
> Please fix this one up so we can merge soon.

I'll fix it today, soz for not doing it yesterday just got a bit sick and I was too tired to do anything.

**Comment by fantoro** [comment]
> When zooming, the center of the image or the cropped portion doesn't stay at the center of the image pane.

I'm not sure what would be the best way to do that though. I'll try to experiment and maybe I'll come up with something.

> Zooming in decelerates while zoom out accelerates. You probably want to multiply/divide rather than adding/subtracting when updating the zoom factor.

I tried multiplying/dividing before but it sort of felt weird for me, although I might've used wrong values so I'll try again.

**Comment by fantoro** [comment]
> > When zooming, the center of the image or the cropped portion doesn't stay at the center of the image pane.
> 
> I'm not sure what would be the best way to do that though. I'll try to experiment and maybe I'll come up with something.

Ok yeah I can't figure out how to do it to be honest. I tried experimenting with ratios but it didn't really it work.

File: source/AllesView.cpp
Comment (owenca) [inline source/AllesView.cpp]: I am merely suggesting a more gradual zooming here. If it doesn't work as well, feel free to keep yours.
```suggestion
		zoom *= ychange > 0 ? 0.8 : 1.25;
```

**Comment by owenca** [comment]
> > > When zooming, the center of the image or the cropped portion doesn't stay at the center of the image pane.
> > 
> > 
> > I'm not sure what would be the best way to do that though. I'll try to experiment and maybe I'll come up with something.
> 
> Ok yeah I can't figure out how to do it to be honest. I tried experimenting with ratios but it didn't really work.

Can you create a new issue for this? We can deal with it in the future.

**Comment by fantoro** [comment]
> [...] 
> Can you create a new issue for this? We can deal with it in the future.

Should I do that now or after this PR gets merged?

**Comment by owenca** [comment]
Let's wait until after it's merged. Thanks!

File: source/WindowPeek.cpp
Comment (owenca) [inline source/WindowPeek.cpp]: Shouldn't these be changed to multiply/divide by 1.25?

Comment (fantoro) [reply source/WindowPeek.cpp]: Oh shoot, I forgot about this one because I focused so much on the scroll wheel zooming.

**Review by owenca — APPROVED** [review]

**Pull request merged**
